web/app/js/components/TapQueryForm.jsx

```diff
@@ -183,14 +181,14 @@ class TapQueryForm extends React.Component {
 
       this.setState(state);
       this.props.updateQuery(state.query);
+      this.handleUrlUpdate(state.query);
     };
   }
 
   // Each time state.query is updated, this method calls the equivalent
-  // onChange method to reflect the update in url query params. These onChange
-  // methods are automatically added to props by react-url-query.
-  handleUrlUpdate = (name, formVal) => {
-    this.props[`onChange${_upperFirst(name)}`](formVal);
+  // method to reflect the update in url query params.
+  handleUrlUpdate = query => {
+    this.props.setQuery({ ...query });
```

Comment:
`setQuery` by default only does partial updates (with setting replaceIn as the default). So we don't need now to call it every time that one `query` value is updated.